node/src/components/linear_chain.rs

```diff
@@ -242,32 +246,31 @@ impl<I> LinearChain<I> {
         I: Display + Send + 'static,
     {
         let mut effects = Effects::new();
-        let block_hash = block.hash();
+        let mut known_signatures = self.signature_cache.get_known_signatures(block_hash);
```

Comment:
This is still wrong. We must not collect any signature we already know of but only those that are still pending.